### ISSUE TITLE
`after` hook is not triggered if the test fails

### ISSUE THREAD

**Issue by Nouxx** [webdriverio/webdriverio#7261]
Environment (please complete the following information):

- WebdriverIO version: "^7.3.0"
- Mode: WDIO Testrunner
- If WDIO Testrunner, running sync/async: sync: "^7.0.8"
- Node.js version: node:14.16-alpine3.10 (Docker)
- NPM version: according to node version
- Additional wdio packages used (if applicable): none

**Config of WebdriverIO**
https://gist.github.com/Nouxx/c4d564e4e627ccf4d6d10fc310b494d3

**Describe the bug**
From this snippet of `wdio.conf.js` (see above), I'm trying to implement the `after` hook which has wrong documentation according to #6356n thanks herr @christian-bromann.
The hook is triggered at the end of any execution as expected, which is fine, but it is only launched if there is no error in the actual execution
```js
    after: function (result, capabilities, specs) {
        console.log("after hook")
        var index = devices.findIndex(device => device.udid == capabilities['appium:udid'])
        devices[index].available = true
        console.log(`device ${capabilities['appium:udid']} is released`)
        console.log("DEBUG DEVICES")
        console.log(devices)
    },
```
In the meantime, 'beforeSession' entry can be found in my logs

**To Reproduce**
N/A

**Expected behavior**
`after` hook to be triggered at every end of execution, whatever the result is

**Log**
https://gist.github.com/Nouxx/43f5cb92a3928ba22cb389828a90c13b

**Comment by christian-bromann** [comment]
> The hook is triggered at the end of any execution as expected, which is fine, but it is only launched if there is no error in the actual execution

I don't understand, so the hook is being executed but then isn't?

**Comment by Nouxx** [comment]
Yes sorry, it is not triggered if the test throw an error

In my conf file, I'm using a console.log to debug
`console.log("after hook")`

As you can see in the log, 3 test are run, no `after hook` log entry is present

**Comment by christian-bromann** [comment]
Fixed in #7290

**Comment by Nouxx** [comment]
Thanks a lot! :)
